OKTA-497465 update MFA to authenticator enrollment policy for OIE (#3537)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/policy/index.md

#### Authenticator enrollment policy settings example

<ApiLifecycle access="ie" />

> **Note:** Policy settings are included only for those authenticators that are enabled.

```json
   "settings": {
     "type": "AUTHENTICATORS",
     "authenticators": [
       {
         "key": "security_question",
         "enroll": {
           "self": "OPTIONAL"
         }
       },
       {
         "key": "okta_phone",
         "enroll": {
           "self": "OPTIONAL"
         }
       }
     ]
   }
```

### Policy Settings data

<ApiLifecycle access="ie" />

| Parameter                                                                        | Description                           | Data Type                                                                    | Required | Default   |
| ---                                                                              | ---                                   | ---                                                                          | ---      | ---       |
| authenticators | List of authenticator policy settings | Array of [Policy Authenticator object](#policy-authenticator-object) | No       |           |

> | factors                                                                          | Factor policy settings. This parameter is for Classic Engine MFA Enrollment policies that have migrated to Identity Engine but haven't converted to using authenticators yet. Factors and authenticators are mutually exclusive in an authenticator enrollment policy. When a policy is updated to use authenticators, the factors are removed.               | [Policy Factors Configuration object](#policy-factors-configuration-object)  | No       |           |

Reviewer: Is this table describing the object structure above? If so I don't see "Factors" in the code snippet.

Author: `factors` is optional and mutually exclusive to `authenticators`. You will only see one or the other in the `settings` structure.

Reviewer: okay

| self      | Requirements for the user-initiated enrollment | `NOT_ALLOWED`, `OPTIONAL`, or `REQUIRED` | Yes      | `NOT_ALLOWED` |

### Policy conditions

The following conditions may be applied to authenticator enrollment policies:

* [People Condition](#people-condition-object)

* [Network Condition](#network-condition-object)

* [Application and App Instance Condition](#application-and-app-instance-condition-object)

### Authenticator Rules Action data

#### Authenticator enrollment rules actions example

```json
  "actions": {
    "enroll": {
      "self": "CHALLENGE"
    }
  },
```

#### Rules Actions Enroll object

| Parameter | Description                                                                                               | Data Type                       | Required | Default |
| ---       | ---                                                                                                       | ---                             | ---      | ---     |
| self      | Should the User be enrolled the first time they `LOGIN`, the next time they are `CHALLENGE`d, or `NEVER`? | `CHALLENGE`, `LOGIN` or `NEVER` | Yes      | N/A     |

### Rules conditions

> The following conditions may be applied to the Rules associated with the authenticator enrollment policy:

Reviewer 2: ```suggestion
You can apply the following conditions to the Rules associated with the authenticator enrollment policy:
```
